subset/switches/src/main/java/switchtest/SwitchTest.java

```diff
@@ -0,0 +1,156 @@
+            + "Speed is less than or equal to 10 mbps\n";
+      }
+    }
+    return "RESULT fail connection.port_speed Cannot detect current speed\n";
+  }
+
+  protected String testDuplex(InterfaceResponse interfaceResponse) {
+    String duplex = interfaceResponse.getDuplex();
+    if (duplex != null) {
+      if (duplex.equals("full")) {
+        return "RESULT pass connection.port_duplex Full duplex mode detected\n";
+      } else {
+        return "RESULT fail connection.port_duplex Incorrect duplex mode set\n";
+      }
+    }
+    return "RESULT fail connection.port_duplex Cannot detect duplex mode\n";
+  }
+
+  protected String testPower(PowerResponse powerResponse) {
+    String testResults = "";
+    if (!deviceConfigPoeEnabled) {
+      testResults += "RESULT skip poe.power This test is disabled\n";
+      testResults += "RESULT skip poe.negotiation This test is disabled\n";
+      testResults += "RESULT skip poe.support This test is disabled\n";
+      return testResults;
+    }
+
+    POEStatus poeStatus = powerResponse.getPoeStatus();
+    // Determine PoE power test result
+    if (poeStatus == POEStatus.ON) {
+      if (powerResponse.getMaxPowerConsumption() >= powerResponse.getCurrentPowerConsumption()) {
+        testResults += "RESULT pass poe.power PoE is applied to device\n";
+      } else {
+        testResults += "RESULT fail poe.power device wattage exceeds the max wattage.\n";
+      }
+    } else if (poeStatus == POEStatus.OFF) {
+      testResults += "RESULT fail poe.power No poE is applied\n";
+    } else if (poeStatus == POEStatus.FAULT) {
+      testResults += "RESULT fail poe.power Device detection "
+          + "or a powered device is in a faulty state\n";
+    } else {
+      testResults += "RESULT fail poe.power A powered device is detected, "
+          + "but no PoE is available, or the maximum wattage exceeds the "
+          + "detected powered-device maximum.\n";
+    }
+
+    // Determine PoE auto negotiation result
+    if (powerResponse.getPoeNegotiation() == POENegotiation.NEGOTIATION_ENABLED) {
+      testResults += "RESULT pass poe.negotiation PoE auto-negotiated successfully\n";
+    } else {
+      testResults += "RESULT fail poe.negotiation Incorrect privilege for negotiation\n";
+    }
+
+    // Determine PoE support result
+    if (powerResponse.getPoeSupport() == POESupport.ENABLED) {
+      testResults += "RESULT pass poe.support PoE supported and enabled\n";
+    } else {
+      testResults +=
+          "RESULT fail poe.support The switch does not support PoE or it is disabled\n";
+    }
+
+    return testResults;
+  }
+
+  protected void writeReport(String loginReport) {
+    try {
+      if (debug) {
+        System.out.println("report:" + loginReport);
+      }
+
+      String[] directory = reportFilename.split("/");
+
+      File dir = new File(directory[directory.length - 2]);
+      if (!dir.exists()) {
+        dir.mkdirs();
```

Comment:
check for exists() is redundant for mkdirs() -- it will silently do nothing if the directory already exists. (at least I'm 99% sure)